### PR TITLE
shfmt: fix behavior of -f flag for file arguments

### PR DESCRIPTION
**Pull request by tbcs**
Ensure that invoking shfmt -f on files yields the same result as when
invoked with the directory containing the files. In other words, the
output of `shfmt -f` should only contain files which have been detected
as shell files. This changes shfmt's behavior whereby previously, given
the following file tree:

    bin
    ├── README.md
    └── script.sh

running `shfmt -f bin` would output:

    bin/script.sh

but running `shfmt -f bin/script.sh bin/README.md` would output:

    bin/script.sh
    bin/README.md

As of this commit the output for both cases is consistent.

### CODE REVIEW

File: cmd/shfmt/main_test.go [hunk 1]
```diff
@@ -70,9 +71,9 @@ var walkTests = []struct {
 	{None, false, "ext.other", " foo"},
 	{None, false, "ext-shebang.other", "#!/bin/sh\n foo"},
 	{None, false, "shebang-nospace", "#!/bin/envsh\n foo"},
-	{None, false, filepath.Join(".git", "ext.sh"), " foo"},
-	{None, false, filepath.Join(".svn", "ext.sh"), " foo"},
-	{None, false, filepath.Join(".hg", "ext.sh"), " foo"},
+	{Skip, false, filepath.Join("foo", ".git", "ext.sh"), " foo"},
```
Comment (mvdan) [inline cmd/shfmt/main_test.go]: why the change with `"foo"`? that seems unnecessary

Comment (tbcs) [reply cmd/shfmt/main_test.go]: That was just by mistake, removed now.
